uktrade/great-cms
# KLS-683 - Export academy signup (existing aventri user)

**Context**
This PR adds the first version of the export academy signup flow. In this version, the user is already registered for the export academy via Aventri, and has received a unique link where their email address will be pre-populated and they just need to add a password.

The backend implementation of the unique link has not yet been created, so the view currently relies on a registration uuid in the query parameters of the URL. Once the backend has been implemented this will be updated accordingly.

**Reusing IOO code**
This implementation reuses the logic used by international online offer's signup view which uses django, rather than the main signup flow using react. Currently

As there is a lot of common code between the IOO and EA signup flows, I have created a signup mixin and a verification code mixin in the `sso` app which both IOO and EA now use.

**To test**
- Make sure you have an EA registration for your chosen email in your local db
- Make sure you have directory-forms-api, directory-sso and directory-sso-proxy running locally
- Go to `export-academy/signup?registration-id=<your_registration_id>`
- Your email address should be pre-populated
- You should be able to go through the signup and verification code process and end up at `export-academy/events`
- If you already have an account in sso for your chosen email, try putting a `+1` before the @ sign

**Screenshots**
![Screenshot 2023-06-21 at 10 37 24](https://github.com/uktrade/great-cms/assets/22460823/c7cb404c-6bac-4109-9fc3-8857efe50db7)

![Screenshot 2023-06-21 at 10 38 03](https://github.com/uktrade/great-cms/assets/22460823/797b1c43-4d72-484d-9207-538e85c8bcc1)

### Workflow
- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-683

### Reviewing help


### Housekeeping


### Merging
